shards/aggregate.go

```diff
@@ -47,6 +49,11 @@ func (c *collectSender) Send(r *zoekt.SearchResult) {
 	if len(r.Files) > 0 {
 		c.aggregate.Files = append(c.aggregate.Files, r.Files...)
 
+		zoekt.SortFiles(c.aggregate.Files)
+		if max := c.opts.MaxDocDisplayCount; max > 0 && max < len(c.aggregate.Files) {
+			c.aggregate.Files = c.aggregate.Files[:max]
+		}
```

Comment:
I think my original reasoning against doing this was doing N sorts vs one final sort was more CPU usage. I think it is,  but I think the tradeoff here is worth it, especially since the sort algorithm in go I think already handles runs of incrementing so is likely quite fast.

But yeah, because we know max and that both inputs are sorted we could _very_ efficiently implement this if we wanted to. I'm almost tempted to implement it now because its some of my favourite sort of code to write, but I'll stave off the temptation and let someone else have fun.

Reply:
I'll implement this now, I was also thinking we should just do a sorted merge here.

Reply:
I'll implement this now, I was also thinking we should just do a sorted merge here. I also wasn't sure if it would help in practice though, since the sort algorithm should efficiently handle runs (as you mention).